gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -865,4 +1016,23 @@ public static Builder builder() {
    * @throws StorageException upon failure
    */
   List<Boolean> delete(BlobId... blobIds);
+
+  /**
+   * Returns a {@link BlobRewriter} object for the provided {@code rewriteRequest}. If source and
+   * destination objects share the same location and storage class the source blob is copied with a
+   * single call of {@link BlobRewriter#copyChunk()}, regardless of the {@link
+   * RewriteRequest#maxBytesRewrittenPerCall} parameter. If source and destination have different
+   * location or storage class multiple RPC calls might be needed depending on blob's size.
+   * <p>
+   * Example usage of blob rewriter:
+   * <pre>    {@code BlobRewriter rewriter = service.rewriter(rewriteRequest);}
+   *    {@code while(!rewriter.isDone()) {
+   *       rewriter.copyChunk();
+   *   }}
+   * </pre>
+   *
+   * @throws StorageException upon failure
+   * @see <a href="https://cloud.google.com/storage/docs/json_api/v1/objects/rewrite">Rewrite</a>
+   */
+  BlobRewriter rewriter(RewriteRequest rewriteRequest);
```

Comment:
Should we call it `rewrite`?

I wonder if we really need to have both copy and rewrite methods?

One option could be to make the copy do the rewrite, so basically change copy to return a `CopyResponse` (similar
to the `BlobRewriter`) with `isDone`, `nextChunk` and `blobInfo`.

We could also make `blobInfo` a blocking call until complete (or have a more explicit method for it).

/cc @BrandonY @mfschwartz